havenapi/actions/keycloak.go

```diff
@@ -0,0 +1,90 @@
+package actions
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+)
+
+type token struct {
+	Token string `json:"access_token"`
+}
+
+// For example purposes
+type users struct {
+	LastName string `json:"email"`
+	ID       int    `json:"id"`
+}
+
+var adminToken = token{}
+
+// Example env vars (KC_HOST=http://localhost | KC_PORT=2015)
+var adminUser = os.Getenv("KC_ADMIN")
+var adminPw = os.Getenv("KC_PW")
+var keycloakHost = os.Getenv("KC_HOST") + ":" + os.Getenv("KC_PORT")
+var getTokenURL = "/auth/realms/master/protocol/openid-connect/token"
+var getUsersURL = "/auth/admin/realms/havendev/users"
+
+// KeycloakGetToken grabs the token for the admin api.
+func KeycloakGetToken() bool {
+
+	form := url.Values{
+		"username":   {adminUser},
+		"password":   {adminPw},
+		"client_id":  {"admin-cli"},
+		"grant_type": {"password"},
+	}
+
+	body := bytes.NewBufferString(form.Encode())
+	resp, err := http.Post(
+		keycloakHost+getTokenURL,
+		"application/x-www-form-urlencoded",
+		body,
+	)
+	if err != nil {
+		return false
+	}
+	defer resp.Body.Close()
+
+	bodyByte, err := ioutil.ReadAll(resp.Body)
+
+	if err != nil {
+		return false
```

Comment:
I think we should be using log.Error in any error clause, similar to https://github.com/kindlyops/mappamundi/blob/2f9101f31577dbf57974b7c27e49f57b77a3c974/havenapi/actions/files.go#L26. Also I think it would be good go style to return error rather than bool, probably adding context to the error using `fmt.Errorf` or `Wrap` as described in https://dave.cheney.net/2016/04/27/dont-just-check-errors-handle-them-gracefully.